### ISSUE TITLE
.source_attributions tensor returns float64 with integrated gradients method

### ISSUE THREAD

**Issue by Leukas** [inseq-team/inseq#156]
## 🐛 Bug Report

The output tensor .source_attributions in FeatureAttributionSequenceOutput is of type float64 when using "integrated_gradients" method, rather than the expected float32. 

## 🔬 How To Reproduce

1. Run any model using "integrated_gradients" method
2. Inspect the dtype of out.sequence_attributions[0].source_attributions

### Code sample
```
import inseq
model = inseq.load_model("Helsinki-NLP/opus-mt-en-fr", "integrated_gradients")
out = model.attribute(
  "The developer argued with the designer because her idea cannot be implemented.",
  n_steps=100
)
print(out.sequence_attributions[0].source_attributions.dtype)
```
### Environment

Python 3.8.16

## 📈 Expected behavior

The dtype should be float32. 
## 📎 Additional context

Other methods ("saliency", "input_x_gradient", "deeplift") return float32.
Interestingly, "discretized_integrated_gradients" also returns float32, but "layer_integrated_gradients" returns float64.

**Comment by Leukas** [comment]
Additionally the .target_attributions tensor is also float64.

**Comment by gsarti** [comment]
Thanks for reporting, should be fixed in #157!
